src/Makefile

```diff
@@ -4,23 +4,33 @@
 #    SPDX-License-Identifier: GPL-2.0-or-later
 #==============================================================================
 
-CC = gcc
+CC       ?= gcc
+CFLAGS   ?= -g -Wall -Werror
+CPPFLAGS ?=
+LDFLAGS  ?=
+INCLUDES  = -I../inc/
 
-COPTS = -g -Wall -Werror
-CFLAGS = -I../inc/.
+PREFIX   ?= /usr/local
+BINDIR   ?= $(PREFIX)/bin
+DATADIR  ?= $(PREFIX)/share
 
-DEPS = cst_signer.h cfg_parser.h mkimage_helper.h
-SRCS = cst_signer.c cfg_parser.c mkimage_helper.c fdt.o
+SRCS = cst_signer.c cfg_parser.c mkimage_helper.c fdt.c
 
-.PHONY: all clean
+all: cst-signer
 
-all: cst_signer fdt.o
+%.o : %.c
+	$(CC) -c $(INCLUDES) $(CFLAGS) $(LDFLAGS) $(CPPFLAGS) $< -o $@
 
-fdt.o: fdt.c
-	$(CC) -c -w -o $@ $< $(CFLAGS)
+cst-signer: $(SRCS:.c=.o)
+	$(CC) $(INCLUDES) $(CFLAGS) $(LDFLAGS) $(CPPFLAGS) -o $@ $(SRCS:.c=.o)
 
-cst_signer: cst_signer.c fdt.o
-	$(CC) $(COPTS) $(CFLAGS) -o $@ $(SRCS)
+install: cst-signer
+	install -D -m 0755 cst-signer $(DESTDIR)/$(BINDIR)/cst-signer
+	install -D -m 0755 -t $(DESTDIR)$(DATADIR)/doc/cst-signer \
```

Comment:
I do not understand the usage of doc folder here. maybe its something I am not familiar with possibly autotools related?

Reply:
The .sample configuration files go to /usr/share/doc/ as
```
-rw-r--r-- root/root       230 2023-09-08 16:41 ./usr/share/doc/cst-signer/csf_ahab.cfg.sample
-rw-r--r-- root/root       430 2023-09-08 16:41 ./usr/share/doc/cst-signer/csf_hab4.cfg.sample
```
for target packages. Here for example I generated:
```
[0] ~/src/oe/oel/master/build/tmp/deploy/ipk/cortexa53-crypto/cst-signer-dev_0.3+0+7fa50daf32-r0_cortexa53-crypto.ipk [0]
[1] ~/src/oe/oel/master/build/tmp/deploy/ipk/cortexa53-crypto/cst-signer-doc_0.3+0+7fa50daf32-r0_cortexa53-crypto.ipk [1]
[2] ~/src/oe/oel/master/build/tmp/deploy/ipk/cortexa53-crypto/cst-signer_0.3+0+7fa50daf32-r0_cortexa53-crypto.ipk [2]
[3] ~/src/oe/oel/master/build/tmp/deploy/ipk/cortexa53-crypto/cst-signer-src_0.3+0+7fa50daf32-r0_cortexa53-crypto.ipk [3]
[4] ~/src/oe/oel/master/build/tmp/deploy/ipk/cortexa53-crypto/cst-signer-dbg_0.3+0+7fa50daf32-r0_cortexa53-crypto.ipk [4]
```
I understand that you use those files as templates, but in this particular case, it should be done in the native recipe. In my recipe I did this as well.